assets/src/dashboard/app/views/savedTemplates/index.js

```diff
@@ -17,19 +17,142 @@
+import { InfiniteScroller, Layout } from '../../../components';
+import { StoriesPropType } from '../../../types';
+import { UnitsProvider } from '../../../../edit-story/units';
+import { reshapeTemplateObject } from '../../api/useTemplateApi';
+import { useConfig } from '../../config';
+import { TransformProvider } from '../../../../edit-story/components/transform';
+import FontProvider from '../../font/fontProvider';
+
+function Header({ search, stories, view, sort }) {
+  const listBarLabel = sprintf(
+    /* translators: %s: number of templates */
+    _n(
+      '%s total template',
+      '%s total templates',
+      stories.length,
+      'web-stories'
+    ),
+    stories.length
+  );
+  return (
+    <Layout.Squishable>
+      <PageHeading
+        defaultTitle={__('Saved Templates', 'web-stories')}
+        searchPlaceholder={__('Search Templates', 'web-stories')}
+        stories={stories}
+        handleTypeaheadChange={search.setKeyword}
+        typeaheadValue={search.keyword}
+      />
+      <BodyViewOptions
+        listBarLabel={listBarLabel}
+        layoutStyle={view.style}
+        currentSort={sort.value}
+        handleSortChange={sort.set}
+        sortDropdownAriaLabel={__(
+          'Choose sort option for display',
+          'web-stories'
+        )}
+      />
+    </Layout.Squishable>
+  );
+}
+
+function Content({ stories, view, page }) {
+  return (
+    <Layout.Scrollable>
+      <FontProvider>
+        <TransformProvider>
+          <UnitsProvider pageSize={view.pageSize}>
+            <BodyWrapper>
+              <StoryGridView
+                stories={stories}
+                centerActionLabel={__('View', 'web-stories')}
+                bottomActionLabel={__('Use template', 'web-stories')}
+                isTemplate
+              />
+              <InfiniteScroller
+                allDataLoadedMessage={__('No more templates.', 'web-stories')}
+                isLoading={false}
+                canLoadMore={false}
+                onLoadMore={page.requestNextPage}
+              />
+            </BodyWrapper>
+          </UnitsProvider>
+        </TransformProvider>
+      </FontProvider>
+    </Layout.Scrollable>
+  );
+}
 
 function SavedTemplates() {
+  const config = useConfig();
+  const { search, view, page, sort } = useStoryView({
+    filters: [],
+    totalPages: 1,
+  });
+
+  /**
+   * A placeholder to just have template data in the view for now.
+   */
+  const mockTemplates = useRef(
+    getAllTemplates(config).map(reshapeTemplateObject(false))
+  );
+
   return (
-    <div>
-      <ViewHeader>{__('Saved Templates', 'web-stories')}</ViewHeader>
-    </div>
+    <Layout.Provider>
+      <Header
+        view={view}
+        search={search}
+        stories={mockTemplates.current}
+        sort={sort}
+      />
+      <Content
+        view={view}
+        page={page}
+        sort={sort}
+        stories={mockTemplates.current}
+      />
+    </Layout.Provider>
   );
```

Comment:
By dividing the view into two functions it becomes easier to compose and test each part rather than a sequence of useMemo hooks.